// @affinda/affinda/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreClient = require('@azure/core-client');

const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
const $host = {
    parameterPath: "$host",
    mapper: {
        serializedName: "$host",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
const limit = {
    parameterPath: "limit",
    mapper: {
        defaultValue: 300,
        constraints: {
            InclusiveMinimum: 1
        },
        serializedName: "limit",
        type: {
            name: "Number"
        }
    }
};
const offset = {
    parameterPath: "offset",
    mapper: {
        constraints: {
            InclusiveMinimum: 0
        },
        serializedName: "offset",
        type: {
            name: "Number"
        }
    }
};
const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "multipart/form-data",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
const file = {
    parameterPath: ["options", "file"],
    mapper: {
        serializedName: "file",
        type: {
            name: "Stream"
        }
    }
};
const identifier = {
    parameterPath: ["options", "identifier"],
    mapper: {
        serializedName: "identifier",
        type: {
            name: "String"
        }
    }
};
const fileName = {
    parameterPath: ["options", "fileName"],
    mapper: {
        serializedName: "fileName",
        type: {
            name: "String"
        }
    }
};
const url = {
    parameterPath: ["options", "url"],
    mapper: {
        serializedName: "url",
        type: {
            name: "String"
        }
    }
};
const wait = {
    parameterPath: ["options", "wait"],
    mapper: {
        defaultValue: "true",
        serializedName: "wait",
        type: {
            name: "String"
        }
    }
};
const language = {
    parameterPath: ["options", "language"],
    mapper: {
        serializedName: "language",
        type: {
            name: "String"
        }
    }
};
const expiryTime = {
    parameterPath: ["options", "expiryTime"],
    mapper: {
        serializedName: "expiryTime",
        type: {
            name: "String"
        }
    }
};
const accept1 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String"
        }
    }
};
const identifier1 = {
    parameterPath: "identifier",
    mapper: {
        serializedName: "identifier",
        required: true,
        type: {
            name: "String"
        }
    }
};
const redactHeadshot = {
    parameterPath: ["options", "redactHeadshot"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactHeadshot",
        type: {
            name: "String"
        }
    }
};
const redactPersonalDetails = {
    parameterPath: ["options", "redactPersonalDetails"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactPersonalDetails",
        type: {
            name: "String"
        }
    }
};
const redactWorkDetails = {
    parameterPath: ["options", "redactWorkDetails"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactWorkDetails",
        type: {
            name: "String"
        }
    }
};
const redactEducationDetails = {
    parameterPath: ["options", "redactEducationDetails"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactEducationDetails",
        type: {
            name: "String"
        }
    }
};
const redactReferees = {
    parameterPath: ["options", "redactReferees"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactReferees",
        type: {
            name: "String"
        }
    }
};
const redactLocations = {
    parameterPath: ["options", "redactLocations"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactLocations",
        type: {
            name: "String"
        }
    }
};
const redactDates = {
    parameterPath: ["options", "redactDates"],
    mapper: {
        defaultValue: "true",
        serializedName: "redactDates",
        type: {
            name: "String"
        }
    }
};
const resumeFormat = {
    parameterPath: "resumeFormat",
    mapper: {
        serializedName: "resumeFormat",
        required: true,
        type: {
            name: "String"
        }
    }
};

const GetAllDocumentsResults = {
    type: {
        name: "Composite",
        className: "GetAllDocumentsResults",
        modelProperties: {
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            next: {
                serializedName: "next",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            previous: {
                serializedName: "previous",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            results: {
                serializedName: "results",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Meta"
                        }
                    }
                }
            }
        }
    }
};
const Meta = {
    type: {
        name: "Composite",
        className: "Meta",
        modelProperties: {
            identifier: {
                serializedName: "identifier",
                required: true,
                nullable: true,
                type: {
                    name: "String"
                }
            },
            fileName: {
                serializedName: "fileName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            ready: {
                serializedName: "ready",
                required: true,
                type: {
                    name: "Boolean"
                }
            },
            readyDt: {
                serializedName: "readyDt",
                nullable: true,
                type: {
                    name: "DateTime"
                }
            },
            failed: {
                serializedName: "failed",
                required: true,
                type: {
                    name: "Boolean"
                }
            },
            expiryTime: {
                serializedName: "expiryTime",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const RequestError = {
    type: {
        name: "Composite",
        className: "RequestError",
        modelProperties: {
            detail: {
                serializedName: "detail",
                required: true,
                type: {
                    name: "String"
                }
            },
            statusCode: {
                serializedName: "statusCode",
                required: true,
                type: {
                    name: "Number"
                }
            }
        }
    }
};
const Resume = {
    type: {
        name: "Composite",
        className: "Resume",
        modelProperties: {
            data: {
                serializedName: "data",
                type: {
                    name: "Composite",
                    className: "ResumeData"
                }
            },
            meta: {
                serializedName: "meta",
                type: {
                    name: "Composite",
                    className: "Meta"
                }
            },
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorModel"
                }
            }
        }
    }
};
const ResumeData = {
    type: {
        name: "Composite",
        className: "ResumeData",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "Composite",
                    className: "ResumeDataName"
                }
            },
            phoneNumbers: {
                serializedName: "phoneNumbers",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            websites: {
                serializedName: "websites",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            emails: {
                serializedName: "emails",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            dateOfBirth: {
                serializedName: "dateOfBirth",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "Composite",
                    className: "Location"
                }
            },
            objective: {
                serializedName: "objective",
                type: {
                    name: "String"
                }
            },
            languages: {
                serializedName: "languages",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            summary: {
                serializedName: "summary",
                type: {
                    name: "String"
                }
            },
            totalYearsExperience: {
                serializedName: "totalYearsExperience",
                type: {
                    name: "Number"
                }
            },
            headShot: {
                serializedName: "headShot",
                nullable: true,
                type: {
                    name: "ByteArray"
                }
            },
            education: {
                serializedName: "education",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResumeDataEducationItem"
                        }
                    }
                }
            },
            profession: {
                serializedName: "profession",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            linkedin: {
                serializedName: "linkedin",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            workExperience: {
                serializedName: "workExperience",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResumeDataWorkExperienceItem"
                        }
                    }
                }
            },
            skills: {
                serializedName: "skills",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResumeDataSkillsItem"
                        }
                    }
                }
            },
            certifications: {
                serializedName: "certifications",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            publications: {
                serializedName: "publications",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            referees: {
                serializedName: "referees",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResumeDataRefereesItem"
                        }
                    }
                }
            },
            sections: {
                serializedName: "sections",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResumeDataSectionsItem"
                        }
                    }
                }
            },
            isResumeProbability: {
                serializedName: "isResumeProbability",
                type: {
                    name: "Number"
                }
            },
            rawText: {
                serializedName: "rawText",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ResumeDataName = {
    type: {
        name: "Composite",
        className: "ResumeDataName",
        modelProperties: {
            raw: {
                serializedName: "raw",
                type: {
                    name: "String"
                }
            },
            first: {
                serializedName: "first",
                type: {
                    name: "String"
                }
            },
            last: {
                serializedName: "last",
                type: {
                    name: "String"
                }
            },
            middle: {
                serializedName: "middle",
                type: {
                    name: "String"
                }
            },
            title: {
                serializedName: "title",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const Location = {
    type: {
        name: "Composite",
        className: "Location",
        modelProperties: {
            formatted: {
                serializedName: "formatted",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            postalCode: {
                serializedName: "postalCode",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            country: {
                serializedName: "country",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            rawInput: {
                serializedName: "rawInput",
                required: true,
                type: {
                    name: "String"
                }
            },
            streetNumber: {
                serializedName: "streetNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            street: {
                serializedName: "street",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            apartmentNumber: {
                serializedName: "apartmentNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            city: {
                serializedName: "city",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ResumeDataEducationItem = {
    type: {
        name: "Composite",
        className: "ResumeDataEducationItem",
        modelProperties: {
            organization: {
                serializedName: "organization",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            accreditation: {
                serializedName: "accreditation",
                type: {
                    name: "Composite",
                    className: "ResumeDataEducationItemAccreditation"
                }
            },
            grade: {
                serializedName: "grade",
                type: {
                    name: "Composite",
                    className: "ResumeDataEducationItemGrade"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "Composite",
                    className: "Location"
                }
            },
            dates: {
                serializedName: "dates",
                type: {
                    name: "Composite",
                    className: "ResumeDataEducationItemDates"
                }
            }
        }
    }
};
const ResumeDataEducationItemAccreditation = {
    type: {
        name: "Composite",
        className: "ResumeDataEducationItemAccreditation",
        modelProperties: {
            education: {
                serializedName: "education",
                type: {
                    name: "String"
                }
            },
            inputStr: {
                serializedName: "inputStr",
                type: {
                    name: "String"
                }
            },
            matchStr: {
                serializedName: "matchStr",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            educationLevel: {
                serializedName: "educationLevel",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ResumeDataEducationItemGrade = {
    type: {
        name: "Composite",
        className: "ResumeDataEducationItemGrade",
        modelProperties: {
            raw: {
                serializedName: "raw",
                type: {
                    name: "String"
                }
            },
            metric: {
                serializedName: "metric",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ResumeDataEducationItemDates = {
    type: {
        name: "Composite",
        className: "ResumeDataEducationItemDates",
        modelProperties: {
            completionDate: {
                serializedName: "completionDate",
                type: {
                    name: "Date"
                }
            },
            isCurrent: {
                serializedName: "isCurrent",
                type: {
                    name: "Boolean"
                }
            },
            startDate: {
                serializedName: "startDate",
                nullable: true,
                type: {
                    name: "Date"
                }
            }
        }
    }
};
const ResumeDataWorkExperienceItem = {
    type: {
        name: "Composite",
        className: "ResumeDataWorkExperienceItem",
        modelProperties: {
            jobTitle: {
                serializedName: "jobTitle",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            organization: {
                serializedName: "organization",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            location: {
                serializedName: "location",
                type: {
                    name: "Composite",
                    className: "Location"
                }
            },
            jobDescription: {
                serializedName: "jobDescription",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            dates: {
                serializedName: "dates",
                type: {
                    name: "Composite",
                    className: "ResumeDataWorkExperienceItemDates"
                }
            }
        }
    }
};
const ResumeDataWorkExperienceItemDates = {
    type: {
        name: "Composite",
        className: "ResumeDataWorkExperienceItemDates",
        modelProperties: {
            startDate: {
                serializedName: "startDate",
                nullable: true,
                type: {
                    name: "Date"
                }
            },
            endDate: {
                serializedName: "endDate",
                nullable: true,
                type: {
                    name: "Date"
                }
            },
            monthsInPosition: {
                serializedName: "monthsInPosition",
                nullable: true,
                type: {
                    name: "Number"
                }
            },
            isCurrent: {
                serializedName: "isCurrent",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
const ResumeDataSkillsItem = {
    type: {
        name: "Composite",
        className: "ResumeDataSkillsItem",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            lastUsed: {
                serializedName: "lastUsed",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            numberOfMonths: {
                serializedName: "numberOfMonths",
                nullable: true,
                type: {
                    name: "Number"
                }
            },
            type: {
                serializedName: "type",
                type: {
                    name: "String"
                }
            },
            sources: {
                serializedName: "sources",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ResumeDataSkillsPropertiesItemsItem"
                        }
                    }
                }
            }
        }
    }
};
const ResumeDataSkillsPropertiesItemsItem = {
    type: {
        name: "Composite",
        className: "ResumeDataSkillsPropertiesItemsItem",
        modelProperties: {
            section: {
                serializedName: "section",
                type: {
                    name: "String"
                }
            },
            position: {
                serializedName: "position",
                nullable: true,
                type: {
                    name: "Number"
                }
            }
        }
    }
};
const ResumeDataRefereesItem = {
    type: {
        name: "Composite",
        className: "ResumeDataRefereesItem",
        modelProperties: {
            name: {
                serializedName: "name",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            email: {
                serializedName: "email",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            number: {
                serializedName: "number",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ResumeDataSectionsItem = {
    type: {
        name: "Composite",
        className: "ResumeDataSectionsItem",
        modelProperties: {
            sectionType: {
                serializedName: "sectionType",
                type: {
                    name: "String"
                }
            },
            bbox: {
                constraints: {
                    MinItems: 4,
                    MaxItems: 4
                },
                serializedName: "bbox",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Number"
                        }
                    }
                }
            },
            pageIndex: {
                serializedName: "pageIndex",
                type: {
                    name: "Number"
                }
            },
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ErrorModel = {
    type: {
        name: "Composite",
        className: "ErrorModel",
        modelProperties: {
            errorCode: {
                serializedName: "errorCode",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            errorDetail: {
                serializedName: "errorDetail",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const RedactedResume = {
    type: {
        name: "Composite",
        className: "RedactedResume",
        modelProperties: {
            data: {
                serializedName: "data",
                type: {
                    name: "Composite",
                    className: "RedactedResumeData"
                }
            },
            meta: {
                serializedName: "meta",
                type: {
                    name: "Composite",
                    className: "Meta"
                }
            },
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorModel"
                }
            }
        }
    }
};
const RedactedResumeData = {
    type: {
        name: "Composite",
        className: "RedactedResumeData",
        modelProperties: {
            redactedPdf: {
                serializedName: "redactedPdf",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const Paths1UtuacyResumeFormatsGetResponses200ContentApplicationJsonSchema = {
    type: {
        name: "Composite",
        className: "Paths1UtuacyResumeFormatsGetResponses200ContentApplicationJsonSchema",
        modelProperties: {
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            next: {
                serializedName: "next",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            previous: {
                serializedName: "previous",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            results: {
                serializedName: "results",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Get200ApplicationJsonPropertiesItemsItem"
                        }
                    }
                }
            }
        }
    }
};
const Get200ApplicationJsonPropertiesItemsItem = {
    type: {
        name: "Composite",
        className: "Get200ApplicationJsonPropertiesItemsItem",
        modelProperties: {
            identifier: {
                serializedName: "identifier",
                required: true,
                nullable: true,
                type: {
                    name: "String"
                }
            },
            formatFile: {
                serializedName: "formatFile",
                required: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const ReformattedResume = {
    type: {
        name: "Composite",
        className: "ReformattedResume",
        modelProperties: {
            data: {
                serializedName: "data",
                type: {
                    name: "Composite",
                    className: "ReformattedResumeData"
                }
            },
            meta: {
                serializedName: "meta",
                type: {
                    name: "Composite",
                    className: "Meta"
                }
            },
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorModel"
                }
            }
        }
    }
};
const ReformattedResumeData = {
    type: {
        name: "Composite",
        className: "ReformattedResumeData",
        modelProperties: {
            reformattedFile: {
                serializedName: "reformattedFile",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const GetAllInvoicesResults = {
    type: {
        name: "Composite",
        className: "GetAllInvoicesResults",
        modelProperties: {
            count: {
                serializedName: "count",
                type: {
                    name: "Number"
                }
            },
            next: {
                serializedName: "next",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            previous: {
                serializedName: "previous",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            results: {
                serializedName: "results",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "Meta"
                        }
                    }
                }
            }
        }
    }
};
const Invoice = {
    type: {
        name: "Composite",
        className: "Invoice",
        modelProperties: {
            data: {
                serializedName: "data",
                type: {
                    name: "Composite",
                    className: "InvoiceData"
                }
            },
            meta: {
                serializedName: "meta",
                type: {
                    name: "Composite",
                    className: "Meta"
                }
            },
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorModel"
                }
            }
        }
    }
};
const InvoiceData = {
    type: {
        name: "Composite",
        className: "InvoiceData",
        modelProperties: {
            invoiceDate: {
                serializedName: "invoiceDate",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            invoiceOrderDate: {
                serializedName: "invoiceOrderDate",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentDateDue: {
                serializedName: "paymentDateDue",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentAmountBase: {
                serializedName: "paymentAmountBase",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentAmountTax: {
                serializedName: "paymentAmountTax",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentAmountTotal: {
                serializedName: "paymentAmountTotal",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentAmountPaid: {
                serializedName: "paymentAmountPaid",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentAmountDue: {
                serializedName: "paymentAmountDue",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            invoiceNumber: {
                serializedName: "invoiceNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            invoicePurchaseOrderNumber: {
                serializedName: "invoicePurchaseOrderNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            supplierBusinessNumber: {
                serializedName: "supplierBusinessNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            customerNumber: {
                serializedName: "customerNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            customerBusinessNumber: {
                serializedName: "customerBusinessNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            paymentReference: {
                serializedName: "paymentReference",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bankAccountNumber: {
                serializedName: "bankAccountNumber",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            supplierVAT: {
                serializedName: "supplierVAT",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            customerVAT: {
                serializedName: "customerVAT",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bpayBillerCode: {
                serializedName: "bpayBillerCode",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bpayReference: {
                serializedName: "bpayReference",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bankSortCode: {
                serializedName: "bankSortCode",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bankIban: {
                serializedName: "bankIBAN",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bankSwift: {
                serializedName: "bankSwift",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            bankBSB: {
                serializedName: "bankBSB",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            customerContactName: {
                serializedName: "customerContactName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            customerCompanyName: {
                serializedName: "customerCompanyName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            supplierCompanyName: {
                serializedName: "supplierCompanyName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            customerBillingAddress: {
                serializedName: "customerBillingAddress",
                type: {
                    name: "Composite",
                    className: "Location"
                }
            },
            customerDeliveryAddress: {
                serializedName: "customerDeliveryAddress",
                type: {
                    name: "Composite",
                    className: "Location"
                }
            },
            supplierAddress: {
                serializedName: "supplierAddress",
                type: {
                    name: "Composite",
                    className: "Location"
                }
            },
            customerPhoneNumber: {
                serializedName: "customerPhoneNumber",
                type: {
                    name: "String"
                }
            },
            supplierPhoneNumber: {
                serializedName: "supplierPhoneNumber",
                type: {
                    name: "String"
                }
            },
            supplierFax: {
                serializedName: "supplierFax",
                type: {
                    name: "String"
                }
            },
            customerEmail: {
                serializedName: "customerEmail",
                type: {
                    name: "String"
                }
            },
            supplierEmail: {
                serializedName: "supplierEmail",
                type: {
                    name: "String"
                }
            },
            supplierWebsite: {
                serializedName: "supplierWebsite",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const Paths7EskthResumesPostRequestbodyContentMultipartFormDataSchema = {
    type: {
        name: "Composite",
        className: "Paths7EskthResumesPostRequestbodyContentMultipartFormDataSchema",
        modelProperties: {
            file: {
                serializedName: "file",
                type: {
                    name: "Stream"
                }
            },
            identifier: {
                serializedName: "identifier",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            fileName: {
                serializedName: "fileName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            url: {
                serializedName: "url",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            wait: {
                defaultValue: "true",
                serializedName: "wait",
                type: {
                    name: "String"
                }
            },
            language: {
                serializedName: "language",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            expiryTime: {
                serializedName: "expiryTime",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const Paths8DdhfcRedactedResumesPostRequestbodyContentMultipartFormDataSchema = {
    type: {
        name: "Composite",
        className: "Paths8DdhfcRedactedResumesPostRequestbodyContentMultipartFormDataSchema",
        modelProperties: {
            file: {
                serializedName: "file",
                type: {
                    name: "Stream"
                }
            },
            identifier: {
                serializedName: "identifier",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            fileName: {
                serializedName: "fileName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            url: {
                serializedName: "url",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            language: {
                serializedName: "language",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            wait: {
                defaultValue: "true",
                serializedName: "wait",
                type: {
                    name: "String"
                }
            },
            redactHeadshot: {
                defaultValue: "true",
                serializedName: "redactHeadshot",
                type: {
                    name: "String"
                }
            },
            redactPersonalDetails: {
                defaultValue: "true",
                serializedName: "redactPersonalDetails",
                type: {
                    name: "String"
                }
            },
            redactWorkDetails: {
                defaultValue: "true",
                serializedName: "redactWorkDetails",
                type: {
                    name: "String"
                }
            },
            redactEducationDetails: {
                defaultValue: "true",
                serializedName: "redactEducationDetails",
                type: {
                    name: "String"
                }
            },
            redactReferees: {
                defaultValue: "true",
                serializedName: "redactReferees",
                type: {
                    name: "String"
                }
            },
            redactLocations: {
                defaultValue: "true",
                serializedName: "redactLocations",
                type: {
                    name: "String"
                }
            },
            redactDates: {
                defaultValue: "true",
                serializedName: "redactDates",
                type: {
                    name: "String"
                }
            },
            expiryTime: {
                serializedName: "expiryTime",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};
const PathsYzn84IReformattedResumesPostRequestbodyContentMultipartFormDataSchema = {
    type: {
        name: "Composite",
        className: "PathsYzn84IReformattedResumesPostRequestbodyContentMultipartFormDataSchema",
        modelProperties: {
            file: {
                serializedName: "file",
                type: {
                    name: "Stream"
                }
            },
            identifier: {
                serializedName: "identifier",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            fileName: {
                serializedName: "fileName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            url: {
                serializedName: "url",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            language: {
                serializedName: "language",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            resumeFormat: {
                serializedName: "resumeFormat",
                required: true,
                type: {
                    name: "String"
                }
            },
            wait: {
                defaultValue: "true",
                serializedName: "wait",
                type: {
                    name: "String"
                }
            }
        }
    }
};
const Paths1BwrvmkInvoicesPostRequestbodyContentMultipartFormDataSchema = {
    type: {
        name: "Composite",
        className: "Paths1BwrvmkInvoicesPostRequestbodyContentMultipartFormDataSchema",
        modelProperties: {
            file: {
                serializedName: "file",
                type: {
                    name: "Stream"
                }
            },
            identifier: {
                serializedName: "identifier",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            fileName: {
                serializedName: "fileName",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            url: {
                serializedName: "url",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            wait: {
                defaultValue: "true",
                serializedName: "wait",
                type: {
                    name: "String"
                }
            },
            language: {
                serializedName: "language",
                nullable: true,
                type: {
                    name: "String"
                }
            },
            expiryTime: {
                serializedName: "expiryTime",
                nullable: true,
                type: {
                    name: "String"
                }
            }
        }
    }
};

var Mappers = /*#__PURE__*/Object.freeze({
    __proto__: null,
    GetAllDocumentsResults: GetAllDocumentsResults,
    Meta: Meta,
    RequestError: RequestError,
    Resume: Resume,
    ResumeData: ResumeData,
    ResumeDataName: ResumeDataName,
    Location: Location,
    ResumeDataEducationItem: ResumeDataEducationItem,
    ResumeDataEducationItemAccreditation: ResumeDataEducationItemAccreditation,
    ResumeDataEducationItemGrade: ResumeDataEducationItemGrade,
    ResumeDataEducationItemDates: ResumeDataEducationItemDates,
    ResumeDataWorkExperienceItem: ResumeDataWorkExperienceItem,
    ResumeDataWorkExperienceItemDates: ResumeDataWorkExperienceItemDates,
    ResumeDataSkillsItem: ResumeDataSkillsItem,
    ResumeDataSkillsPropertiesItemsItem: ResumeDataSkillsPropertiesItemsItem,
    ResumeDataRefereesItem: ResumeDataRefereesItem,
    ResumeDataSectionsItem: ResumeDataSectionsItem,
    ErrorModel: ErrorModel,
    RedactedResume: RedactedResume,
    RedactedResumeData: RedactedResumeData,
    Paths1UtuacyResumeFormatsGetResponses200ContentApplicationJsonSchema: Paths1UtuacyResumeFormatsGetResponses200ContentApplicationJsonSchema,
    Get200ApplicationJsonPropertiesItemsItem: Get200ApplicationJsonPropertiesItemsItem,
    ReformattedResume: ReformattedResume,
    ReformattedResumeData: ReformattedResumeData,
    GetAllInvoicesResults: GetAllInvoicesResults,
    Invoice: Invoice,
    InvoiceData: InvoiceData,
    Paths7EskthResumesPostRequestbodyContentMultipartFormDataSchema: Paths7EskthResumesPostRequestbodyContentMultipartFormDataSchema,
    Paths8DdhfcRedactedResumesPostRequestbodyContentMultipartFormDataSchema: Paths8DdhfcRedactedResumesPostRequestbodyContentMultipartFormDataSchema,
    PathsYzn84IReformattedResumesPostRequestbodyContentMultipartFormDataSchema: PathsYzn84IReformattedResumesPostRequestbodyContentMultipartFormDataSchema,
    Paths1BwrvmkInvoicesPostRequestbodyContentMultipartFormDataSchema: Paths1BwrvmkInvoicesPostRequestbodyContentMultipartFormDataSchema
});

class AffindaAPIContext extends coreClient.ServiceClient {
    /**
     * Initializes a new instance of the AffindaAPIContext class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param options The parameter options
     */
    constructor(credentials, options) {
        if (credentials === undefined) {
            throw new Error("'credentials' cannot be null");
        }
        // Initializing default values for options
        if (!options) {
            options = {};
        }
        const defaults = {
            requestContentType: "application/json; charset=utf-8",
            credential: credentials
        };
        const packageDetails = `azsdk-js-affinda/0.3.0`;
        const userAgentPrefix = options.userAgentOptions && options.userAgentOptions.userAgentPrefix
            ? `${options.userAgentOptions.userAgentPrefix} ${packageDetails}`
            : `${packageDetails}`;
        if (!options.credentialScopes) {
            options.credentialScopes = ["https://management.azure.com/.default"];
        }
        const optionsWithDefaults = Object.assign(Object.assign(Object.assign({}, defaults), options), { userAgentOptions: {
                userAgentPrefix
            }, baseUri: options.endpoint || "https://api.affinda.com/v1" });
        super(optionsWithDefaults);
        // Assigning values to Constant parameters
        this.$host = options.$host || "https://api.affinda.com/v1";
    }
}

class AffindaAPI extends AffindaAPIContext {
    /**
     * Initializes a new instance of the AffindaAPI class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param options The parameter options
     */
    constructor(credentials, options) {
        super(credentials, options);
    }
    /**
     * Returns all the resume summaries for that user, limited to 300 per page.
     * @param options The options parameters.
     */
    getAllResumes(options) {
        return this.sendOperationRequest({ options }, getAllResumesOperationSpec);
    }
    /**
     * Uploads a resume for parsing.
     * When successful, returns an `identifier` in the response for subsequent use with the
     * [/resumes/{identifier}](#operation/getResume) endpoint to check processing status and retrieve
     * results.
     * @param options The options parameters.
     */
    createResume(options) {
        return this.sendOperationRequest({ options }, createResumeOperationSpec);
    }
    /**
     * Returns all the parse results for that resume if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the resume via the
     * [/resumes](#operation/createResume) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getResumeOperationSpec);
    }
    /**
     * Deletes the specified resume from the database
     * @param identifier Resume identifier
     * @param options The options parameters.
     */
    deleteResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteResumeOperationSpec);
    }
    /**
     * Returns all the redacted resume information for that resume
     * @param options The options parameters.
     */
    getAllRedactedResumes(options) {
        return this.sendOperationRequest({ options }, getAllRedactedResumesOperationSpec);
    }
    /**
     * Uploads a resume for redacting.
     * @param options The options parameters.
     */
    createRedactedResume(options) {
        return this.sendOperationRequest({ options }, createRedactedResumeOperationSpec);
    }
    /**
     * Returns all the redaction results for that resume if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the resume via the
     * [/redacted_resumes](#operation/createRedactedResume) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getRedactedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getRedactedResumeOperationSpec);
    }
    /**
     * Deletes the specified resume from the database
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    deleteRedactedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteRedactedResumeOperationSpec);
    }
    /**
     * Returns all the resume formats
     * @param options The options parameters.
     */
    getAllResumeFormats(options) {
        return this.sendOperationRequest({ options }, getAllResumeFormatsOperationSpec);
    }
    /**
     * Returns all the reformatted resume information for that resume
     * @param options The options parameters.
     */
    getAllReformattedResumes(options) {
        return this.sendOperationRequest({ options }, getAllReformattedResumesOperationSpec);
    }
    /**
     * Upload a resume for reformatting.
     * @param resumeFormat Identifier of the format used
     * @param options The options parameters.
     */
    createReformattedResume(resumeFormat, options) {
        return this.sendOperationRequest({ resumeFormat, options }, createReformattedResumeOperationSpec);
    }
    /**
     * Returns all the reformatting results for that resume if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the resume via the
     * [/reformatted_resumes](#operation/createReformattedResume) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getReformattedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getReformattedResumeOperationSpec);
    }
    /**
     * Delete the specified resume from the database
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    deleteReformattedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteReformattedResumeOperationSpec);
    }
    /**
     * Returns all the invoice summaries for that user, limited to 300 per page.
     * @param options The options parameters.
     */
    getAllInvoices(options) {
        return this.sendOperationRequest({ options }, getAllInvoicesOperationSpec);
    }
    /**
     * Uploads an invoice for parsing.
     * When successful, returns an `identifier` in the response for subsequent use with the
     * [/invoices/{identifier}](#operation/getInvoice) endpoint to check processing status and retrieve
     * results.
     * @param options The options parameters.
     */
    createInvoice(options) {
        return this.sendOperationRequest({ options }, createInvoiceOperationSpec);
    }
    /**
     * Returns all the parse results for that invoice if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the invoice via the
     * [/invoices](#operation/createInvoice) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getInvoice(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getInvoiceOperationSpec);
    }
    /**
     * Delete the specified invoice from the database
     * @param identifier Invoice identifier
     * @param options The options parameters.
     */
    deleteInvoice(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteInvoiceOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getAllResumesOperationSpec = {
    path: "/resumes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: GetAllDocumentsResults
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    queryParameters: [limit, offset],
    urlParameters: [$host],
    headerParameters: [accept],
    serializer
};
const createResumeOperationSpec = {
    path: "/resumes",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Resume
        },
        201: {
            bodyMapper: Resume
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    formDataParameters: [
        file,
        identifier,
        fileName,
        url,
        wait,
        language,
        expiryTime
    ],
    urlParameters: [$host],
    headerParameters: [contentType, accept1],
    serializer
};
const getResumeOperationSpec = {
    path: "/resumes/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Resume
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const deleteResumeOperationSpec = {
    path: "/resumes/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const getAllRedactedResumesOperationSpec = {
    path: "/redacted_resumes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: GetAllDocumentsResults
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    queryParameters: [limit, offset],
    urlParameters: [$host],
    headerParameters: [accept],
    serializer
};
const createRedactedResumeOperationSpec = {
    path: "/redacted_resumes",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: RedactedResume
        },
        201: {
            bodyMapper: RedactedResume
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    formDataParameters: [
        file,
        identifier,
        fileName,
        url,
        wait,
        language,
        expiryTime,
        redactHeadshot,
        redactPersonalDetails,
        redactWorkDetails,
        redactEducationDetails,
        redactReferees,
        redactLocations,
        redactDates
    ],
    urlParameters: [$host],
    headerParameters: [contentType, accept1],
    serializer
};
const getRedactedResumeOperationSpec = {
    path: "/redacted_resumes/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: RedactedResume
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const deleteRedactedResumeOperationSpec = {
    path: "/redacted_resumes/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const getAllResumeFormatsOperationSpec = {
    path: "/resume_formats",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Paths1UtuacyResumeFormatsGetResponses200ContentApplicationJsonSchema
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    queryParameters: [limit, offset],
    urlParameters: [$host],
    headerParameters: [accept],
    serializer
};
const getAllReformattedResumesOperationSpec = {
    path: "/reformatted_resumes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: GetAllDocumentsResults
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    queryParameters: [limit, offset],
    urlParameters: [$host],
    headerParameters: [accept],
    serializer
};
const createReformattedResumeOperationSpec = {
    path: "/reformatted_resumes",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: ReformattedResume
        },
        201: {
            bodyMapper: ReformattedResume
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    formDataParameters: [
        file,
        identifier,
        fileName,
        url,
        wait,
        language,
        resumeFormat
    ],
    urlParameters: [$host],
    headerParameters: [contentType, accept1],
    serializer
};
const getReformattedResumeOperationSpec = {
    path: "/reformatted_resumes/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: ReformattedResume
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const deleteReformattedResumeOperationSpec = {
    path: "/reformatted_resumes/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const getAllInvoicesOperationSpec = {
    path: "/invoices",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: GetAllInvoicesResults
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    queryParameters: [limit, offset],
    urlParameters: [$host],
    headerParameters: [accept],
    serializer
};
const createInvoiceOperationSpec = {
    path: "/invoices",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Invoice
        },
        201: {
            bodyMapper: Invoice
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    formDataParameters: [
        file,
        identifier,
        fileName,
        url,
        wait,
        language,
        expiryTime
    ],
    urlParameters: [$host],
    headerParameters: [contentType, accept1],
    serializer
};
const getInvoiceOperationSpec = {
    path: "/invoices/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Invoice
        },
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};
const deleteInvoiceOperationSpec = {
    path: "/invoices/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: RequestError
        },
        401: {
            bodyMapper: RequestError
        },
        404: {
            bodyMapper: RequestError
        },
        default: {
            bodyMapper: RequestError
        }
    },
    urlParameters: [$host, identifier1],
    headerParameters: [accept],
    serializer
};

class AffindaCredential {
    constructor(token) {
        this.token = token;
    }
    getToken(scopes, options) {
        return Promise.resolve({
            token: this.token,
            expiresOnTimestamp: 0
        });
    }
}

exports.AffindaAPI = AffindaAPI;
exports.AffindaAPIContext = AffindaAPIContext;
exports.AffindaCredential = AffindaCredential;
