// @affinda/affinda/dist-esm/affindaAPI.js
import * as coreClient from "@azure/core-client";
import * as Parameters from "./models/parameters";
import * as Mappers from "./models/mappers";
import { AffindaAPIContext } from "./affindaAPIContext";
export class AffindaAPI extends AffindaAPIContext {
    /**
     * Initializes a new instance of the AffindaAPI class.
     * @param credentials Subscription credentials which uniquely identify client subscription.
     * @param options The parameter options
     */
    constructor(credentials, options) {
        super(credentials, options);
    }
    /**
     * Returns all the resume summaries for that user, limited to 300 per page.
     * @param options The options parameters.
     */
    getAllResumes(options) {
        return this.sendOperationRequest({ options }, getAllResumesOperationSpec);
    }
    /**
     * Uploads a resume for parsing.
     * When successful, returns an `identifier` in the response for subsequent use with the
     * [/resumes/{identifier}](#operation/getResume) endpoint to check processing status and retrieve
     * results.
     * @param options The options parameters.
     */
    createResume(options) {
        return this.sendOperationRequest({ options }, createResumeOperationSpec);
    }
    /**
     * Returns all the parse results for that resume if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the resume via the
     * [/resumes](#operation/createResume) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getResumeOperationSpec);
    }
    /**
     * Deletes the specified resume from the database
     * @param identifier Resume identifier
     * @param options The options parameters.
     */
    deleteResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteResumeOperationSpec);
    }
    /**
     * Returns all the redacted resume information for that resume
     * @param options The options parameters.
     */
    getAllRedactedResumes(options) {
        return this.sendOperationRequest({ options }, getAllRedactedResumesOperationSpec);
    }
    /**
     * Uploads a resume for redacting.
     * @param options The options parameters.
     */
    createRedactedResume(options) {
        return this.sendOperationRequest({ options }, createRedactedResumeOperationSpec);
    }
    /**
     * Returns all the redaction results for that resume if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the resume via the
     * [/redacted_resumes](#operation/createRedactedResume) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getRedactedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getRedactedResumeOperationSpec);
    }
    /**
     * Deletes the specified resume from the database
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    deleteRedactedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteRedactedResumeOperationSpec);
    }
    /**
     * Returns all the resume formats
     * @param options The options parameters.
     */
    getAllResumeFormats(options) {
        return this.sendOperationRequest({ options }, getAllResumeFormatsOperationSpec);
    }
    /**
     * Returns all the reformatted resume information for that resume
     * @param options The options parameters.
     */
    getAllReformattedResumes(options) {
        return this.sendOperationRequest({ options }, getAllReformattedResumesOperationSpec);
    }
    /**
     * Upload a resume for reformatting.
     * @param resumeFormat Identifier of the format used
     * @param options The options parameters.
     */
    createReformattedResume(resumeFormat, options) {
        return this.sendOperationRequest({ resumeFormat, options }, createReformattedResumeOperationSpec);
    }
    /**
     * Returns all the reformatting results for that resume if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the resume via the
     * [/reformatted_resumes](#operation/createReformattedResume) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getReformattedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getReformattedResumeOperationSpec);
    }
    /**
     * Delete the specified resume from the database
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    deleteReformattedResume(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteReformattedResumeOperationSpec);
    }
    /**
     * Returns all the invoice summaries for that user, limited to 300 per page.
     * @param options The options parameters.
     */
    getAllInvoices(options) {
        return this.sendOperationRequest({ options }, getAllInvoicesOperationSpec);
    }
    /**
     * Uploads an invoice for parsing.
     * When successful, returns an `identifier` in the response for subsequent use with the
     * [/invoices/{identifier}](#operation/getInvoice) endpoint to check processing status and retrieve
     * results.
     * @param options The options parameters.
     */
    createInvoice(options) {
        return this.sendOperationRequest({ options }, createInvoiceOperationSpec);
    }
    /**
     * Returns all the parse results for that invoice if processing is completed.
     * The `identifier` is the unique ID returned after POST-ing the invoice via the
     * [/invoices](#operation/createInvoice) endpoint.
     * @param identifier Document identifier
     * @param options The options parameters.
     */
    getInvoice(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, getInvoiceOperationSpec);
    }
    /**
     * Delete the specified invoice from the database
     * @param identifier Invoice identifier
     * @param options The options parameters.
     */
    deleteInvoice(identifier, options) {
        return this.sendOperationRequest({ identifier, options }, deleteInvoiceOperationSpec);
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getAllResumesOperationSpec = {
    path: "/resumes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GetAllDocumentsResults
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    queryParameters: [Parameters.limit, Parameters.offset],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const createResumeOperationSpec = {
    path: "/resumes",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.Resume
        },
        201: {
            bodyMapper: Mappers.Resume
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    formDataParameters: [
        Parameters.file,
        Parameters.identifier,
        Parameters.fileName,
        Parameters.url,
        Parameters.wait,
        Parameters.language,
        Parameters.expiryTime
    ],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.contentType, Parameters.accept1],
    serializer
};
const getResumeOperationSpec = {
    path: "/resumes/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Resume
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteResumeOperationSpec = {
    path: "/resumes/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const getAllRedactedResumesOperationSpec = {
    path: "/redacted_resumes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GetAllDocumentsResults
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    queryParameters: [Parameters.limit, Parameters.offset],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const createRedactedResumeOperationSpec = {
    path: "/redacted_resumes",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.RedactedResume
        },
        201: {
            bodyMapper: Mappers.RedactedResume
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    formDataParameters: [
        Parameters.file,
        Parameters.identifier,
        Parameters.fileName,
        Parameters.url,
        Parameters.wait,
        Parameters.language,
        Parameters.expiryTime,
        Parameters.redactHeadshot,
        Parameters.redactPersonalDetails,
        Parameters.redactWorkDetails,
        Parameters.redactEducationDetails,
        Parameters.redactReferees,
        Parameters.redactLocations,
        Parameters.redactDates
    ],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.contentType, Parameters.accept1],
    serializer
};
const getRedactedResumeOperationSpec = {
    path: "/redacted_resumes/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.RedactedResume
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteRedactedResumeOperationSpec = {
    path: "/redacted_resumes/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const getAllResumeFormatsOperationSpec = {
    path: "/resume_formats",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Paths1UtuacyResumeFormatsGetResponses200ContentApplicationJsonSchema
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    queryParameters: [Parameters.limit, Parameters.offset],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const getAllReformattedResumesOperationSpec = {
    path: "/reformatted_resumes",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GetAllDocumentsResults
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    queryParameters: [Parameters.limit, Parameters.offset],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const createReformattedResumeOperationSpec = {
    path: "/reformatted_resumes",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.ReformattedResume
        },
        201: {
            bodyMapper: Mappers.ReformattedResume
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    formDataParameters: [
        Parameters.file,
        Parameters.identifier,
        Parameters.fileName,
        Parameters.url,
        Parameters.wait,
        Parameters.language,
        Parameters.resumeFormat
    ],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.contentType, Parameters.accept1],
    serializer
};
const getReformattedResumeOperationSpec = {
    path: "/reformatted_resumes/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.ReformattedResume
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteReformattedResumeOperationSpec = {
    path: "/reformatted_resumes/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const getAllInvoicesOperationSpec = {
    path: "/invoices",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.GetAllInvoicesResults
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    queryParameters: [Parameters.limit, Parameters.offset],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.accept],
    serializer
};
const createInvoiceOperationSpec = {
    path: "/invoices",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.Invoice
        },
        201: {
            bodyMapper: Mappers.Invoice
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    formDataParameters: [
        Parameters.file,
        Parameters.identifier,
        Parameters.fileName,
        Parameters.url,
        Parameters.wait,
        Parameters.language,
        Parameters.expiryTime
    ],
    urlParameters: [Parameters.$host],
    headerParameters: [Parameters.contentType, Parameters.accept1],
    serializer
};
const getInvoiceOperationSpec = {
    path: "/invoices/{identifier}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.Invoice
        },
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
const deleteInvoiceOperationSpec = {
    path: "/invoices/{identifier}",
    httpMethod: "DELETE",
    responses: {
        204: {},
        400: {
            bodyMapper: Mappers.RequestError
        },
        401: {
            bodyMapper: Mappers.RequestError
        },
        404: {
            bodyMapper: Mappers.RequestError
        },
        default: {
            bodyMapper: Mappers.RequestError
        }
    },
    urlParameters: [Parameters.$host, Parameters.identifier1],
    headerParameters: [Parameters.accept],
    serializer
};
